src/scikit_build_core/settings/skbuild_model.py

```diff
@@ -27,6 +27,23 @@ def __dir__() -> List[str]:
     return __all__
 
 
+class CMakeSettingsDefine(str):
+    json_schema = Union[str, bool, List[str]]
```

Comment:
Iirc the CI fails because this is not in `__all__`. Not sure if the intention is to be there or not. But at the very least can you add a quick docstring?

Reply:
Added a docstring and marked with `_` prefix to indicate it's a private type.

Reply:
That didn't work, iirc the issue is that `CMakeSettingsDefine` is used in the documentation of `CMakeSettings` and it fails when it's trying to reference it.

I wonder if you could also make it a standard class with `__str__` implementation. It would at least be less confusing compared to having the type change within the constructor of `CMakeSettings`

Reply:
I tried it as a standard class with a `__str__` implementation, but it grew the diff by a good bit. Trying to define it as a dataclass really got the schema code confused, too.

Reply:
Yeah, definitely not a dataclass in this case. Big diffs are not necessarily a problem as long as they are easy to follow. For this one I'm ok with either implementation, but let's see what Henry preferrs as well.

Reply:
Yep, can't be a Dataclass. I think this is fine - where is the type changing? CMakeSettings doesn't have a constructor, it's a Dataclass, and this is how you communicate to it that you want something special?

Reply:
This `__new__` does return a `CMakeDefine`, though? I don’t see how a `str` could be returned.

Reply:
Ah, right, I misread the `super.__new__`. All cool in this case then I guess.